File: solution/ServerExam/Entity/ScantronHistoryEntity.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ServerExam.Entity
{
    public class ScantronHistoryEntity
    {
        [Key]
        [JsonPropertyName("ID")]
        public int ID { get; set; }

        /// <summary>
        /// 标题
        /// </summary>
        [JsonPropertyName("QuestionTitle")]
        public string QuestionTitle { get; set; }

        /// <summary>
        /// 标题代码
        /// </summary>
        [JsonPropertyName("QuestionCode")]
        public string QuestionCode { get; set; }

        /// <summary>
        /// 试题类型
        /// </summary>
        [JsonPropertyName("QuestionType")]
        public int QuestionType { get; set; }

        /// <summary>
        /// 试题状态 1正常 2禁用
        /// </summary>
        [JsonPropertyName("QuestionState")]
        public int QuestionState { get; set; }

        /// <summary>
        /// 是否人工阅卷 1否 2是
        /// </summary>
        [JsonPropertyName("Marking")]
        public int Marking { get; set; }

        /// <summary>
        /// 知识点ID
        /// </summary>
        [JsonPropertyName("KnowledgeID")]
        public int KnowledgeID { get; set; }

        /// <summary>
        /// 试题描述
        /// </summary>
        [JsonPropertyName("Describe")]
        public string Describe { get; set; }

        /// <summary>
        /// 试题附件
        /// </summary>
        [JsonPropertyName("Attachment")]
        public string Attachment { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        [JsonPropertyName("CreateTime")]
        public int CreateTime { get; set; }

        /// <summary>
        /// 更新时间
        /// </summary>
        [JsonPropertyName("UpdateTime")]
        public int UpdateTime { get; set; }

        /// <summary>
        /// 答题卡得分
        /// </summary>
        [JsonPropertyName("Score")]
        public decimal Score { get; set; }

        /// <summary>
        /// 报名ID
        /// </summary>
        [JsonPropertyName("ExamID")]
        public int ExamID { get; set; }

        /// <summary>
        /// 大标题内容
        /// </summary>
        [JsonPropertyName("HeadlineContent")]
        public string HeadlineContent { get; set; }

        public ScantronHistoryEntity()
        {
            this.ID = 0;
            this.QuestionTitle = "";
            this.QuestionCode = "";
            this.QuestionType = 0;
            this.QuestionState = 0;
            this.Marking = 0;
            this.KnowledgeID = 0;
            this.Describe = "";
            this.Attachment = "";
            this.CreateTime = 0;
            this.UpdateTime = 0;
            this.Score = 0;
            this.ExamID = 0;
            this.HeadlineContent = "";
        }
    }
}

File: solution/Entity/ScantronHistoryEntity.py
from Entity.BaseEntity import *


# 历史答题卡
class ScantronHistoryEntity(BaseEntity, BaseORM):
    __tablename__ = 'ScantronHistory'

    QuestionTitle: str = Column(String(65535), comment='标题')
    QuestionCode: str = Column(String(128), comment='试题编码')
    QuestionType: int = Column(INTEGER, comment='试题类型 1单选 2判断 3多选 4填空 5问答 6代码实训 7拖拽题 8连线题')
    Marking: int = Column(INTEGER, comment='人工阅卷 1否 2是')
    KnowledgeID: int = Column(INTEGER, comment='知识点ID')
    Description: str = Column(String(65535), comment='试题描述')
    Attachment: str = Column(String(65535), comment='试题附件')
    UpdateTime: int = Column(INTEGER, comment='更新时间', default=int(time()))
    Score: float = Column(DECIMAL(10, 2), comment='额定分数')
    ExamID: int = Column(INTEGER, comment='报名ID')
    HeadlineContent: str = Column(String(65535), comment='大标题内容')

    def __init__(self):
        super().__init__()

    def SetQuestionTitle(self, Value: str):
        if isinstance(Value, str):
            self.QuestionTitle = Value.strip()

    def SetQuestionCode(self, Value: str):
        if isinstance(Value, str):
            self.QuestionCode = Value.strip()

    def SetQuestionType(self, Value: int):
        if isinstance(Value, int):
            self.QuestionType = Value

    def SetMarking(self, Value: int):
        if isinstance(Value, int):
            self.Marking = Value

    def SetKnowledgeID(self, Value: int):
        if isinstance(Value, int):
            self.KnowledgeID = Value

    def SetDescription(self, Value: str):
        if isinstance(Value, str):
            self.Description = Value.strip()

    def SetAttachment(self, Value: str):
        if isinstance(Value, str):
            self.Attachment = Value.strip()

    def SetUpdateTime(self, Value: int):
        if isinstance(Value, int):
            self.UpdateTime = Value

    def SetScore(self, Value: float):
        if isinstance(Value, float):
            self.Score = Value

    def SetExamID(self, Value: int):
        if isinstance(Value, int):
            self.ExamID = Value

    def SetHeadlineContent(self, Value: str):
        if isinstance(Value, str):
            self.HeadlineContent = Value.strip()